tests/test_picking_batch.py

```diff
@@ -305,3 +307,215 @@ def test14_is_valid_location_dest_failure_unknown_location(self):
         self.assertFalse(
             batch.is_valid_location_dest_id("this location does not exist"),
             "An invalid dest location is wrongly marked as valid")
+
+    def test15_unpickable_item_single_move_line_has_package_success_internal_transfer(self):  # noqa
+        """
+        Tests that the picking is cancelled and an internal transfer is created
+        if a picking type is not specified.
+        """
+        picking, batch = self._create_valid_batch_for_location_tests()
+        move_line_id = picking.move_line_ids[0].id
+        reason = 'missing item'
+        batch.unpickable_item(move_line_id=move_line_id,
+                              reason=reason,
+                              picking_type_id=None)
+        internal_picking = self.package_one.find_move_lines().picking_id  # noqa
+
+        self.assertEqual(picking.state, 'cancel')
```

Comment:
Mind adding super short comments to the assertion calls, indicating what went wrong (expecting an assertion failure - see other tests above)?